### ISSUE TITLE
[Bug?]: GraphQLError: Syntax Error: Unexpected <EOF>. on typeDefinitions `generateMirrorCells`

### ISSUE THREAD

**Issue by chemalopezp** [redwoodjs/redwood#9195]
### What's not working?

When running openTelemetry the server fails with

```
web |   ➜  Local:   http://localhost:8910/
web |   ➜  Network: http://192.168.86.235:8910/
gen | Generating full TypeScript definitions and GraphQL schemas
gen | /Users/chema/inkworks/node_modules/graphql/error/syntaxError.js:15
gen |   return new _GraphQLError.GraphQLError(`Syntax Error: ${description}`, {
gen |          ^
gen | 
gen | GraphQLError: Syntax Error: Unexpected <EOF>.
gen |     at syntaxError (/Users/chema/inkworks/node_modules/graphql/error/syntaxError.js:15:10)
gen |     at Parser.unexpected (/Users/chema/inkworks/node_modules/graphql/language/parser.js:1458:41)
gen |     at Parser.parseDefinition (/Users/chema/inkworks/node_modules/graphql/language/parser.js:212:16)
gen |     at Parser.many (/Users/chema/inkworks/node_modules/graphql/language/parser.js:1511:26)
gen |     at Parser.parseDocument (/Users/chema/inkworks/node_modules/graphql/language/parser.js:122:25)
gen |     at parse (/Users/chema/inkworks/node_modules/graphql/language/parser.js:32:17)
gen |     at parseGqlQueryToAst (/Users/chema/inkworks/node_modules/@redwoodjs/internal/dist/gql.js:18:34)
gen |     at generateMirrorCell (/Users/chema/inkworks/node_modules/@redwoodjs/internal/dist/generate/typeDefinitions.js:113:48)
gen |     at /Users/chema/inkworks/node_modules/@redwoodjs/internal/dist/generate/typeDefinitions.js:90:86
gen |     at Array.map (<anonymous>) {
gen |   path: undefined,
gen |   locations: [ { line: 2, column: 3 } ],
gen |   extensions: [Object: null prototype] {}
gen | }
```

Which points to `generateMirrorCells` on typeDefinitions. Is that related somehow?

### How do we reproduce the bug?

I simply followed the instructions (on both v6.2.1 and v6.2.2) and the error happened after `yarn rw dev`

### Are you interested in working on this?

- [X] I'm interested in working on this

**Comment by chemalopezp** [comment]
Searching a bit closer, I found this issue https://github.com/redwoodjs/redwood/issues/7491 which looks closely related (I'm also using fragments) but mysteriously I can't reproduce it if I removed the openTelemetry settings

**Comment by Josh-Walker-GM** [comment]
Hey @chemalopezp, thanks for raising this. I'll aim to take a look at replicating this later today. Are there any specific steps you can think I might need to do beyond: 
1. Create a new test project
2. Setup opentelemetry
3. Run `yarn rw dev`

**Comment by dthyresson** [comment]
Can you share some code that replicates this?

For example, are you using some interpolation in any web-side query like a fragment that has `${something}` in your `gql` tag?

If so, this is an issue with the way Redwood does GraphQL codegen and if you share the example and we do have a workaround.

Note Fragment support was just added to canary here: https://github.com/redwoodjs/redwood/pulls?q=is%3Apr+is%3Aclosed+fragment

In short, you cannot interpolate fragments like the Apollo and other docs state.

Instead you have to register them with the Apollo Cache. This PR will help you do that.

**Comment by chemalopezp** [comment]
hey @dthyresson and @Josh-Walker-GM,
I was able to reproduce with a scratch repo, just adding a fragment (indeed it has nothing to do with the telemetry). 

I realized the cause is that standard cell generation uses QUERY = gql`...` while it should be functions `QUERY = () => gql`...``, something @jtoar already mentioned when we first discussed using fragments. Because by default cells are not creating this way we eventually had the syntax problem, but once changed into functions it works.

In case you still find it useful this is [the repo I used](https://github.com/chemalopezp/redwood-debug-syntax). I guess you can close this issue, unless we want to repurpose to make standard cells to work with fragments ;)

Thank you!

**Comment by dthyresson** [comment]
This gets resolved by https://github.com/redwoodjs/redwood/pull/9140 which allows the fragment to be registered rather that inlined in the query.
